app/controllers/reviews_controller.rb

```diff
@@ -43,32 +43,31 @@ def downvote
 
   def report
     @review = Review.find(params[:review_id])
-    reason = params[:reason]
-    other_input = params[:other_input]
-    if reason.blank?
-      render json: 'You must select a reason for reporting.',
-             status: :bad_request
-      return
-    end
-
-    url = review_path(@review.id)
+    if (params[:reason].blank?)
+      redirect_to review_path(@review),
+                  notice: 'Unable to complete your request.'
+    else
+      reason = params[:reason]
+      other_input = params[:other_input]
+
+      url = review_path(@review.id)
+      begin
+        UserMailer.report_email(
+          reason,
+          other_input,
+          url,
+          @review.professor_id,
+          @review.course_id
+        )
+          .deliver_now
+      rescue StandardError
+        redirect_to url, notice: 'Unable to complete your request.'
```

Comment:
Why are we `rescue`ing `StandardError` here? This isn't a very readable pattern and doesn't give insight into _why_ the `report_email` function would fail.